### PR TITLE
Add `SearchFilterModal` to integration branch

### PR DESCRIPTION
**Pull request by oisincoveney**
Epic: https://github.com/metabase/metabase/issues/27982
### Description

Adds to #32742 - Search Filter Integration Branch:

- **`SearchFilterModal`**:
	- Displays available filter options (currently, just the `TypeFilter` component), with a future ability to filter the keys based on whatever booleans we decide to put in the future (i.e. for Verified filter)
	- Contains the state for all filters for the search, which is initially populated from the URL when a user accesses a search page
- **`SearchFilterModalFooter`**
	- Breaks out the buttons of the modal into a different component where we can handle the styling and layout separately from the actual functionality of the filters within `SearchFilterModal`

These files will later be used in the `Search Filter for Types` Milestone (#32392). I've put these files in separate PRs to break up the technical feedback process and make sure that nothing gets missed in review or in the testing strategy.

### How to verify

For now, all tests should pass. For now, don't worry about UI/UX, since I've already confirmed the expected behavior with the PD/PM through prototypes, so we will convene in the final review process for this.

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.unit.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  "database",
+  "dataset",
+  "metric",
+  "pulse",
+  "segment",
+  "table",
+];
+
+const TEST_INITIAL_FILTERS: SearchFilters = {
+  type: TEST_TYPES,
```
Comment (WiNloSt) [inline frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.unit.spec.tsx:45]: One thing I'm confusing seeing this the first time is why `type` represents a list of type. Shouldn't this be `types`? Could you help explain this a bit?

File: frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModalFooter.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  onClear: () => void;
+};
+
+export const SearchFilterModalFooter = ({
+  onApply,
+  onCancel,
+  onClear,
+}: SearchFilterModalFooterProps) => {
+  return (
+    <Flex direction={"row"} justify={"space-between"} align={"center"}>
```
Comment (WiNloSt) [inline frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModalFooter.tsx:18]: You should use just a plain string when you can

```suggestion
    <Flex direction="row" justify="space-between" align="center">
```

File: frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  setIsOpen,
+  value,
+  onChangeFilters,
+}: {
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+  value: SearchFilters;
+  onChangeFilters: (filters: SearchFilters) => void;
+}) => {
+  const [output, setOutput] = useState<SearchFilters>(value);
```
Comment (WiNloSt) [inline frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.tsx:31]: I can understand the reason behind `output` but the name itself doesn't help readers understand what it is. Could/should this be renamed to something along the line of `searchFilters`/`setSearchFilters` (using the type as their name)?

File: frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  const applyFilters = () => {
+    onChangeFilters(output);
+    setIsOpen(false);
+  };
+
+  // we can use this field to control which filters are available
+  // - we can enable the 'verified' filter here
+  const availableFilters: FilterTypeKeys[] = useMemo(() => {
+    return [SearchFilterKeys.Type];
```
Comment (WiNloSt) [inline frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.tsx:54]: Correct me if I'm wrong, but are we using the type as a value here? i.e. is this returning
```js
["type"]
```

https://github.com/metabase/metabase/blob/6c9345abceb9e8f9e7d69cb86ec86c922e53b248/frontend/src/metabase/search/types.ts#L17-L19

**Pull request merged**